java/lang/String/String.cpp

```diff
@@ -845,4 +850,170 @@ boolean String::operator>=(const String &target) const {
 		return false;
 	}
 	return true;
+}
+
+String String::print(const String &format, short value)
+{
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if(length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, int value)
+{
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if(length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, long value)
+{
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if(length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, unsigned short value)
+{
+    String result;
+    char buffer[256] = {0};
+    const int length = snprintf(buffer, sizeof(buffer), format.toString(), value);
+    if(length > 0)
+        result = String(buffer, length);
+    return result;
+}
+
+String String::print(const String &format, unsigned int value)
+{

```

Comment:
{ should almost always be at the end of the previous line. We follow the google code style for C++.